src/utils/notifications.ts

```diff
@@ -0,0 +1,64 @@
+import { LocalNotifications } from "@capacitor/local-notifications";
+
+import { getNotificationDates } from "../state/CalculationLogics";
+import { Cycle } from "../data/ClassCycle";
+
+export const requestPermission = () => {
+  LocalNotifications.requestPermissions()
```

Comment:
Set this function as async and use await instead
The same for the functions below